clang/test/Driver/sycl-cuda-arch-macro.cpp

```diff
@@ -1,9 +1,13 @@
 // Verify the __CUDA_ARCH__ macro has not been defined when offloading SYCL on NVPTX
 // RUN: %clangxx -E -dM -fsycl -fsycl-targets=nvptx64-nvidia-cuda -Xsycl-target-backend --offload-arch=sm_80 -nocudalib -fno-sycl-libspirv %s 2>&1 \
 // RUN: | FileCheck --check-prefix=CHECK-CUDA-ARCH-MACRO %s
+// Verify the __CUDA_ARCH__ macro has not been defined when offloading SYCL on AMDGPU
+// RUN: %clangxx -E -dM -fsycl -fsycl-targets=amdgcn-amd-amdhsa -Xsycl-target-backend --offload-arch=gfx90a -nogpulib -fno-sycl-libspirv %s 2>&1 \
+// RUN: | FileCheck --check-prefix=CHECK-CUDA-ARCH-MACRO %s
 // CHECK-CUDA-ARCH-MACRO-NOT: #define __CUDA_ARCH__ {{[0-9]+}}
 
 // Verify that '-fcuda-is-device' is not supplied when offloading SYCL on NVPTX
+// NOTE: AMDGPU targets, i.e. "amdgcn-amd-amdhsa" may rely on "fcuda-is-device"
```

Comment:
NIT: Maybe add a test case to verify that `fcuda-is-device` is passed when offloading `SYCL` to `AMDGPU` targets?

Reply:
I guess I could but I don't know how important this is. Additionally, there is no behavioural change here to warrant the test but I can totally see your thinking here as well. This clang option isn't really a `SYCL` offload for `AMDGPU` thing, but rather just something `AMDGPU` _might_ be relying on and I am uncertain how much overlap there could be to know if we absolutely do or do not need it for `SYCL` or is just useful for `HIP`. Hence, why I left just a comment, because ultimately what we care about here is to prevent the definition of the `__CUDA_ARCH__` macro.
I am not sure if I managed to confuse you even more here :sweat: . 
@srividya-sundaram Please let me know what you think and if you'd still like to see that test added and I can definitely add it. Thanks!